tests/unit/utils/test_credentials.py

```diff
@@ -1,56 +1,152 @@
         # given
-        cred = credentials.Credentials(username="my_user", password="my_password")
+        cred = credentials.Credentials(url="my_url")
+        is_basic_auth_valid.return_value = False
+        is_api_auth_valid.return_value = False
 
         # when / then
-        self.assertRaisesRegexp(CredentialsException, re.compile("URL"), cred.validate)
+        self.assertRaises(CredentialsException, cred.validate)
+
+    def test_is_basic_auth_valid_return_true_when_no_conflict(self):
+        # given
+        cred = credentials.Credentials(username="my_user", password="my_password", publicKey=None, secretKey=None)
+
+        # when
+        valid = cred.is_basic_auth_valid()
+
+        # then
+        self.assertTrue(valid)
+
+    def test_is_basic_auth_valid_return_false_when_conflict(self):
+        # given
+        cred = credentials.Credentials(username="my_user", password="my_password", publicKey="my_pub", secretKey="my_secret")
+
+        # when
+        valid = cred.is_basic_auth_valid()
+
+        # then
+        self.assertFalse(valid)
+
+    def test_is_basic_auth_valid_return_false_when_no_username(self):
+        # given
+        cred = credentials.Credentials(username=None, password="my_password", publicKey=None, secretKey=None)
+
+        # when
+        valid = cred.is_basic_auth_valid()
+
+        # then
+        self.assertFalse(valid)
+
+    def test_is_basic_auth_valid_return_false_when_no_password(self):
+        # given
+        cred = credentials.Credentials(username="my_user", password=None, publicKey=None, secretKey=None)
+
+        # when
+        valid = cred.is_basic_auth_valid()
+
+        # then
+        self.assertFalse(valid)
+
+    def test_is_api_auth_valid_return_true_when_no_conflict(self):
+        # given
+        cred = credentials.Credentials(username="my_user", password=None, publicKey="my_pub", secretKey="my_secret")
+
+        # when
+        valid = cred.is_api_auth_valid()
+
+        # then
+        self.assertTrue(valid)
+
+    def test_is_api_auth_valid_return_false_when_conflict(self):
+        # given
+        cred = credentials.Credentials(username="my_user", password="my_password", publicKey="my_pub", secretKey="my_secret")
+
+        # when
+        valid = cred.is_api_auth_valid()
+
+        # then
+        self.assertFalse(valid)
+
+    def test_is_api_auth_valid_return_false_when_no_username(self):
+        # given
+        cred = credentials.Credentials(username=None, password=None, publicKey="my_pub", secretKey="my_secret")
+
+        # when
+        valid = cred.is_api_auth_valid()
+
+        # then
+        self.assertFalse(valid)
+
+    def test_is_api_auth_valid_return_false_when_no_public_key(self):
+        # given
+        cred = credentials.Credentials(username="my_user", password=None, publicKey=None, secretKey="my_secret")
+
+        # when
+        valid = cred.is_api_auth_valid()
+
+        # then
+        self.assertFalse(valid)
 
```

Comment:
There is not the case return_false_when_no_secret_key

Reply:
done